modules/tensor_mechanics/include/userobjects/CrystalPlasticityStateVarRateComponentVoce.h

```diff
@@ -0,0 +1,72 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef CRYSTALPLASTICITYSTATEVARRATECOMPONENTVOCE_H
+#define CRYSTALPLASTICITYSTATEVARRATECOMPONENTVOCE_H
+
+#include "CrystalPlasticityStateVarRateComponent.h"
+#include "MooseTypes.h"
+
+class CrystalPlasticityStateVarRateComponentVoce;
+
+template <>
+InputParameters validParams<CrystalPlasticityStateVarRateComponentVoce>();
+
+/**
+ * Phenomenological constitutive model state variable evolution rate component userobject class.
+ */
+class CrystalPlasticityStateVarRateComponentVoce : public CrystalPlasticityStateVarRateComponent
+{
+
+public:
+  CrystalPlasticityStateVarRateComponentVoce(const InputParameters & parameters);
+
+  // computing the slip system hardening rate
+  virtual bool calcStateVariableEvolutionRateComponent(unsigned int qp,
+                                                       std::vector<Real> & val) const;
+
+  // class containgn the avialbale slip systems
```

Comment:
could you make those `///` comments so that doxygen picks those up. Also there are some typos in here :-)